File: men_clothes-backend/shopapp/src/main/java/com/project/shopapp/dtos/CartItemDTO.java
package com.project.shopapp.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class CartItemDTO {
    @JsonProperty("product_id")
    private Long productId;

    @JsonProperty("color_id")
    private Long colorId;

    @JsonProperty("size_id")
    private Long sizeId;

    @JsonProperty("quantity")
    private Integer quantity;
}

File: men_clothes-angular/src/app/dtos/order/cart.item.dto.ts
import { IsNumber } from 'class-validator';

export class CartItemDTO {
    @IsNumber()
    product_id: number;

    color_id?: number;

    size_id?: number;

    @IsNumber()
    quantity: number;

    

    constructor(data: any) {
        this.product_id = data.product_id;
        this.color_id = data.color_id;
        this.size_id = data.size_id;
        this.quantity = data.quantity;     
    }
}
